packages/frontend/src/app/main/main.component.spec.ts

```diff
@@ -126,4 +139,82 @@ describe('MainComponent', () => {
       expect(noRepoText).toBeNull();
     }));
   });
+
+  it('should hide the filters when no repository was found', fakeAsync(() => {
+    component.repositories = [];
+    component.loading = false;
+    component.repoName = '';
+    fixture.detectChanges();
+    tick();
+
+    const filterComponent = fixture.debugElement.query(By.css('app-filters'));
+
+    expect(filterComponent).toBeNull();
+  }));
+
+  it('should hide the filters while the page is loading', fakeAsync(() => {
+    component.repositories = [{} as Repository];
+    component.loading = true;
+    component.repoName = '';
+    fixture.detectChanges();
+    tick();
+
+    const filterComponent = fixture.debugElement.query(By.css('app-filters'));
+
+    expect(filterComponent).toBeNull();
+  }));
+
+  it('should hide the filters when a repository is being searched', fakeAsync(() => {
```

Comment:
@bcoe , @StuartRucker , this is the case where the displayed repositories come from a search.